doc/examples/reconst_ivim.py

```diff
@@ -0,0 +1,193 @@
+"""
+============================================================
+Intravoxel incoherent motion
+============================================================
+The intravoxel incoherent motion (IVIM) model describes diffusion
+and perfusion in the signal acquired with diffusion. The IVIM model
+can be understood as an adaptation of Stejskals and Tanners work (ref)
+on biologic tissue, and was proposed by Le Bihan et al (ref).
+The hypothesis is that two compartments exist: a slow moving compartment,
+where particles diffuse in a Brownian fashion as a consequence of thermal
+energy, and a fast moving compartment (the vascular compartment), where
+blood moves as a consequence of pressure gradient. In this second
+compartment, a pseudo diffusion term (D* ) is introduced that describes
+on a macroscopic level the displacement of the blood elements in an
+assumed randomly laid vascular network. For the perfusion to have a
+physiologic meaning, one expects that D* is greater than D.
+
+The IVIM model expresses the diffusion signal as follows:
+
+ .. math: :
+
+    S(b) = S_{0}[fe ^ {-b*D_star} + (1 - f)e^{-b*D}]
+
+where $\mathbf{b}$ is the gradient value (which is dependent on
+the measurement parameters), $S_0$ is the signal in the absence
+of diffusion gradient sensitization, $\mathbf{f}$ is the perfusion
+fraction, $\mathbf{D}$ is the diffusion coefficient,
+ $\mathbf{D * }$ is the pseudo diffusion constant.
+
+In the following example we show how to fit the IVIM model
+on diffusion  weighted dataset and visualize the perfusion
+and diffusion. First, we import all relevant modules:
+"""
+
+import numpy as np
+from scipy.optimize import leastsq
+import matplotlib.pyplot as plt
+
+from dipy.reconst.ivim import IvimModel, _ivim_error, ivim_function
+from dipy.data.fetcher import read_ivim
+
+"""
+We get an IVIM dataset using dipy's data fetcher read_ivim.
+The image data is an acquired IVIM dataset with 21 b - values
+acquired in 3 directions, registered to each other and the
+directions were averaged into one. So, this dataset is 4D
+where the 4th dimension is the b - values. The fetcher
+read_ivim gets the .nii, .bval and .bvec files.
+"""
+
+img, gtab = read_ivim()
+
+"""
+img contains a nibabel Nifti1Image object(with the data)
+and gtab contains a GradientTable object(information about
+the gradients e.g. b - values and b - vectors). We get the data from img using read_data.
+"""
+
+data = img.get_data()
+print('data.shape (%d, %d, %d, %d)' % data.shape)
+
+"""
+Thus the data is a 256x256x54 volume and the fourth dimension
```

Comment:
=> "the data has 54 slices, with 256-by-256 voxels in each slice. The fourth dimension corresponds to the b-values in the `gtab`"